### PR TITLE
Update AMain.cs

### PR DESCRIPTION
**Pull request by damianday**
Made a start on making it more async friendly in order to improve upload.
This is experimental until tested! but the changes are sane.
What remains to be done is File.ReadAllBytes can block.
Motivation for this pull is the UI locking up and being unresponsive while uploading.

### CODE REVIEW

File: AutoPatcherAdmin/AMain.cs [hunk 1]
```diff
@@ -725,12 +734,18 @@ private byte[] Compress(byte[] raw)
...
+            using (GZipStream gStream = new GZipStream(stream, CompressionMode.Compress, true))
+                gStream.WriteAsync(raw, 0, raw.Length);
+        }
     }
 
     public class FileInformation
     {
         public string FileName; //Relative.
-        public int Length, Compressed;
+        public long Length, Compressed;
```
Comment (damianday) [inline AutoPatcherAdmin/AMain.cs:748]: This would be a breaking change I assume..

**Comment by PMySQLQ** [comment]
When uploading, the client files go to disk D

**Comment by Suprcode** [comment]
> When uploading, the client files go to disk D

The client files for temp upload should be copied to the same directory as you're running the admin patcher from.

**Comment by Suprcode** [comment]
Closing this as no response to updating. Feel free to reopen once the suggested change has been made.

**Pull request closed without merging**